### PR TITLE
fix(playground): avoid `assign to readonly` error on Safari

### LINKED ISSUE #3617: Preview playground seems broken on Safari
Tested on Safari 11.0.2 and iOS 10

For example: https://deploy-preview-3614--prettier.netlify.com/playground/

image

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #3617

### CODE REVIEW

**Comment by ikatyang** [comment]
Got another error:

image

Any idea? 🤔

**Comment by j-f1** [comment]
How did you get that error @ikatyang?

**Comment by ikatyang** [comment]
It works now, tested on Safari 11.0.2 and iOS 10.

The root cause is that the top-level `var crypto = true` is not working since the global `crypto` is readonly, wrapping it in an IIFE so that it won't access to the global `crypto`.

File: scripts/build/build-docs.js [hunk 1]
```diff
@@ -38,12 +39,13 @@ shell.cp(`${prettierPath}/index.js`, `${docs}/index.js`);
...
 );
-shell.sed(
-  "-i",
-  /(var crypto=true;)/,
-  "try{$1}catch(e){}",
-  `${docs}/index.js`
-);
+
+// wrap content with IIFE to avoid `assign to readonly` error on Safari
+(function(filename) {
```
Comment (azz) [inline scripts/build/build-docs.js]: I think you can just set `format` to `iife` in `rollup.docs.config.js`?

Comment (ikatyang) [reply scripts/build/build-docs.js]: We built `lib/index.js` from `rollup.index.config.js`, and we need to adjust a lot to get it worked, IMO this is the easiest fix.

**Pull request merged**